src/ast/SymbolTableBuilder.java

```diff
@@ -176,7 +196,11 @@ private void setEnclosingScopeForThisExpr(Expr e, SymbolTable methodST){
      * the method symbols have a pointer to their own scope's symbol table.
      */
     private void buildProgramSymbolTable() {
+        this.buildSuccessful = true;
         for (ClassDecl classdecl : program.classDecls()) {
+            if (!buildSuccessful) {
+                return;
+            }
```

Comment:
don't we want to do this at the end of the loop?

Reply:
why? if something failed inside the class (field redeclaration, var redeclaration inside a method in the last class we checked, then the whole thing is messed up and we don't need to check for more errors